pkg/internal/examples/user.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/instructor-ai/instructor-go/pkg/instructor"
+	"github.com/instructor-ai/instructor-go/pkg/instructor/modes"
+	openai "github.com/sashabaranov/go-openai"
+)
+
+type Person struct {
+	Name string `json:"name"          jsonschema:"title=the name,description=The name of the person,example=joe,example=lucy"`
```

Comment:
does go have decorators?

Reply:
Not like Python no. But you can wrap anything you want in a struct wrapper to add more functionality to it